### PR TITLE
Removing CLI plugins API documentation

### PR DESCRIPTION
**Pull request by silvin-lubecki**
**- What I did**
Adding depreciation notice for the CLI Plugins API.

**- A picture of a cute animal (not mandatory but encouraged)**

image

### CODE REVIEW

File: docs/deprecated.md [hunk 1]
```diff
@@ -50,6 +50,7 @@ The table below provides an overview of the current status of deprecated feature
 
 Status     | Feature                                                                                                                            | Deprecated | Remove
 -----------|------------------------------------------------------------------------------------------------------------------------------------|------------|------------
+Deprecated | [CLI plugins support](#cli-plugins-support)                                                                                        | v20.03.0   | -
```
Comment (tao12345666333) [inline docs/deprecated.md:53]: Sorry for disturbing you.

Does the deprecated `CLI plugins support` mean not to open this function to users (it is not recommended that users develop their own plugins), or does it mean that its function will be completely removed? For example `docker buildx`?

Comment (thaJeztah) [reply docs/deprecated.md:53]: It won't be removed (at least not in the short term) so that (e.g.) buildx can continue to work.

**Comment by SvenDowideit** [comment]
Can someone please provide more context for this? - if you're going to deprecate something we use (and we do), we'd like to know a reason, if there's a plan to replace it, or whatever....

I _was_ working towards simplifying internal use of more complicated workflows using cli plugins, but this kinda comes out of the blue for me.

I presume there is more context :)

**Comment by chris-crone** [comment]
Hi @SvenDowideit,

I gave some context [here](https://github.com/docker/docker.github.io/issues/10852#issuecomment-656625099) and there is discussion there.

Essentially we do not want to fragment the Docker CLI UX and plugins as they are currently structured will cause that. Paths forward are to upstream features that make sense to be upstreamed or build standalone tools for things that are too niche. We are looking at how to make our platform more extendable and this feedback helps us do so.

**Comment by SvenDowideit** [comment]
mmm, I guess the saddest thing is that there wasn't a GitHub discussion about this - I felt that the CLI plugins were the best step to allow users to prototype ideas to make sure they were worth the work to upstream - and a great place for site specific tooling.

Removing the documentation when the API is deprecated is also not really right - the process (er, used to be) to not remove until after the deprecation period was up - as it makes it hard to debug and fix existing and working plugins - it basically mind-wipes the docs.

**Pull request merged**